jetty-util/src/main/java/org/eclipse/jetty/util/HostPort.java

```diff
@@ -24,9 +26,97 @@
  */
 public class HostPort
 {
+    private static final Logger LOG = LoggerFactory.getLogger(HostPort.class);
     private final String _host;
     private final int _port;
 
+    public static HostPort unsafe(String authority)
+    {
+        if (authority == null)
+            return new HostPort("", 0);
+
+        try
+        {
+            if (authority.isEmpty())
+            {
+                return new HostPort(authority, 0);
+            }
+            else if (authority.charAt(0) == '[')
+            {
+                // ipv6reference
+                int close = authority.lastIndexOf(']');
+                String host = authority.substring(0, close + 1);
+                if (authority.length() > close + 1)
+                {
+                    // ipv6 with port
+                    String strPort = authority.substring(close + 2);
```

Comment:
Shouldn't we still check for a ':' in this case?

Reply:
I added a HostPortTest.testValidAuthorityViaUnsafe() as a way to ensure this logic doesn't change the behavior of valid authorities (if being used).